src/main/java/gregtech/api/recipes/map/MapItemStackNBTIngredient.java

```diff
@@ -48,7 +48,18 @@ public boolean equals(Object obj) {
         }
         if (obj instanceof MapItemStackNBTIngredient) {
             MapItemStackNBTIngredient other = (MapItemStackNBTIngredient) obj;
-            return other.gtRecipeInput.acceptsStack(this.stack);
+            if (this.stack.getItem() != other.stack.getItem()) {
+                return false;
+            }
+            if (this.meta != other.meta) {
+                return false;
+            }
```

Comment:
MapItemStackNBTIngredient compares stacks NBT against the recipe ones using the matcher an condition below

Reply:
I think ALongStringOfNumbers mentioned this because you could use `super.equals()` as this extends `MapItemStackIngredient` and it will check the item and meta part for you so no duplicated code here